### ISSUE TITLE
setting max-cache-ttl to 0 causes slowdown problems

### ISSUE THREAD

**Issue by markkundinger** [lancachenet/lancache-dns#137]
### Describe the issue you are having
with the change from last that set max-cache-ttl to 0, the lancache DNS service is now quite slow.  each lookup can take one or two full seconds, and subsequent checks are not cached at all so they also take a second or two.  Further more, even a downstream pihole will not cache the hits because the TTL was set to zero.

The change was made with commit #131 

of the three changes to the options file:
'forward-only' seems fine, and it accomplishes the goal of that commit.
'max-ncache-ttl' doesn't seem to break anything, although the default bind value is 10800
**'max-cache-ttl' of 0 is the problem**.   Per the docs, the default value is 604800, and zero should not be used because it may cause SERVFAIL errors.

https://bind9.readthedocs.io/en/latest/reference.html#namedconf-statement-max-ncache-ttl

I've personally modified it to 10,000 and it works well.

my recommendation is to leave in the forward only line and strip out the other two.  (barring an enhancement to let the user set the cache values)

### How are you running the container(s)

docker compose, pulled latest.

```

### DNS Configuration

```

version: '2.1'

services:
  dns:
    image: lancachenet/lancache-dns:latest
    env_file: .env
#    restart: unless-stopped
    ports:
      - ${DNS_BIND_IP}:53:53/udp
      - ${DNS_BIND_IP}:53:53/tcp

and in .env

## IP address on the host that the DNS server should bind to
DNS_BIND_IP=192.168.3.204

## DNS Resolution for forwarded DNS lookups

UPSTREAM_DNS=149.112.112.11;9.9.9.11

```

### Output of container(s)

```
I have nothing.
```

**Comment by lmginfra** [comment]
yeah would be really nice to have it configurable via env variable. +1

**Comment by tpill90** [comment]
@VibroAxe This one seems to still be a valid issue.  I've seen this one myself before I moved over to generating dnsmasq mappings for Pihole.

**Comment by VibroAxe** [comment]
@MathewBurnett this was one you originally authored the change to ttl on?

**Comment by MathewBurnett** [comment]
I'll try and take a look at this next week and remember why we did what we did. It's entirely possible we can do better

**Comment by MathewBurnett** [comment]
@markkundinger I think you are likely correct here. We did have some issue with nxdomain items which is how this got in there but i suspect that by the time we solved it we had also settled on forward only as well. Let me do some testing and change it.

**Comment by MathewBurnett** [comment]
changed in https://github.com/lancachenet/lancache-dns/commit/55bc29e286ddaa9ea65f27eef9dae0c023685517
